pinot-server/src/main/java/org/apache/pinot/server/api/resources/SegmentMetadataFetcher.java

```diff
@@ -152,6 +155,24 @@ private static Map<String, String> getColumnIndexes(DataSource dataSource) {
       indexStatus.put(JSON_INDEX, INDEX_AVAILABLE);
     }
 
+    if (Objects.isNull(dataSource.getH3Index())) {
+      indexStatus.put(H3_INDEX, INDEX_NOT_AVAILABLE);
+    } else {
+      indexStatus.put(H3_INDEX, INDEX_AVAILABLE);
+    }
+
+    if (Objects.isNull(dataSource.getFSTIndex())) {
+      indexStatus.put(FST_INDEX, INDEX_NOT_AVAILABLE);
+    } else {
+      indexStatus.put(FST_INDEX, INDEX_AVAILABLE);
+    }
+
+    if (Objects.isNull(dataSource.getTextIndex())) {
+      indexStatus.put(TEXT_INDEX, INDEX_NOT_AVAILABLE);
+    } else {
+      indexStatus.put(TEXT_INDEX, INDEX_AVAILABLE);
+    }
```

Comment:
good catch. but I think this getColumnIndexes() is simply broken after adding the new index-spi support, which allows one to add new index types, not in this hard coded list of index types. Perhaps need to use `IndexService.getInstance().getAllIndexes()` to refactor this method.

cc @gortiz who added index-spi support, to shed more light on this.

Reply:
sgtm

Reply:
Yeah, the solution from @klsince is more generic (and easier to read!). I don't mind if we decide to do that in another PR, but I think it is something we should to soon or otherwise we are going to forget about that until we find another index that is not added 😆